### ISSUE TITLE
[V4] Parameters defined in global stages property not loading when used in custom property

### ISSUE THREAD

**Issue by salemol** [serverless/serverless#12755]
### Issue description

Failed to load parameters defined in global stages property when used in custom property

- serverless.yml:
```
service: my-service

stages:
  default:
    params: ${file(env.yml):default}
  prod:
    params: ${file(env.yml):prod}

provider:
  name: aws
  region: ap-southeast-2
  runtime: dotnet8

custom:
  myCustomParameterName: /product/${self:service}-${param:BranchName}-details
```

- env.yml:
```
default:
  BranchName: "develop"
prod:
  BranchName: "prod"
```

When deploying with serverless version `4.1.19`, the parameters failed to load, and I'm encountering the following error:
![image](https://github.com/user-attachments/assets/6dedbad4-0e89-4233-b67c-1ec259c4bb30)

I attempted to resolve this by copying the content from the `env.yml` file into the `serverless.yml` file, which worked. I also tried using the old `params` top-level property, but the error persisted.

**Comment by czubocha** [comment]
Hey @salemol,

Thank you for raising this issue. The variables in Serverless Framework are loaded simultaneously, so at the moment of resolving `${param:BranchName}`, the parameter isn’t loaded yet since it’s being read from the file. We designed the params block in stages to directly include parameters without needing to load them from an external file using `${file(env.yml):default}`.

Do you have a specific requirement to store parameters in an external file when using the params block in Serverless Framework? If so, we recommend using .env files - [Serverless Framework v4 has built-in support for them](https://www.serverless.com/framework/docs/guides/upgrading-v4#environment-variables-loaded-by-default), which might better suit your needs. If using .env files isn’t feasible, I’d be interested to know more about your setup to understand the need for injecting file content directly into params.

**Comment by matiz-altares** [comment]
Hi,
same problem for me. I'm upgrading from v3 to v4, and had parameters in an external file, because, as @salemol said, it's cleaner and easy to manage. A monolitic serverless.yml become a nightmare when creating complex service, so it is vital to split various parts in external files.

@czubocha do you plan to give the possibility to load parameters from an external file? Thank you

**Comment by czubocha** [comment]
@matiz-altares yes, we’re working on it and will let you know as soon as it’s released.

**Comment by czubocha** [comment]
Hey @salemol @matiz-altares, this has been addressed in version 4.4.3, which is now live. Feel free to try it out and let us know if you run into any issues.
